src/components/UploadingJobsDisplay/UploadingJobsDisplay.js

```diff
@@ -48,8 +48,35 @@ class UploadingJobsDisplay extends Component {
     this.uploadJobs();
   }
 
+  componentDidUpdate() {
+    if (!this.props.files) {
+      return;
+    }
+    this.onLeaveHandler();
+  }
+
   componentWillUnmount() {
     this.cancelFileRemovals();
+    this.resetLeaveThePagePrevention();
+  }
+
+  onLeaveHandler() {
+    const notAbleToLeave = Object.keys(this.props.files).some((file) => {
+      return this.props.files[file].status.toLowerCase() === 'uploading';
+    });
+    // prevent from leaving the page in case of download in progress
```

Comment:
Thanks for refactoring. Not a big deal, but please add empty row above the comment and remove the empty row below it so it would be more clear what comment refers to.

Reply:
Thank, updated